### DISCUSSION TITLE
Transfer Projects Between Organizations

### DISCUSSION THREAD

**Discussion by sundowndev** [community/community#8548] [Projects and Issues]
Hi, we recently moved a repo from one org to another and we still have our product board (using the new beta projects) in the old org remaining. It would be great to be able to transfer a project like so. Is that a feature that would cause too many technical limitations or constraints ?

Thanks.

**Comment by vmyazin** [comment]
I would love to transfer a product board to a new organization I created.

**Reply by ghost** [reply]
vmyazin, great comment!

**Comment by Moshyfawn** [comment]
Possibly related #12908

**Comment by delosemre** [comment]
Hi, you can have a look at the document here. hope it helps.

https://docs.github.com/en/repositories/creating-and-managing-repositories/transferring-a-repository

**Reply by Moshyfawn** [reply]
Thank you for the link.

I already went through these docs and haven't found any info **specific to the Projects** feature. There's a transfer option for a repo, but not for Projects _(the Kanban board feature)_

https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects

**Comment by luispuerto** [comment]
I would also love to know how you can transfer a project between organizations

**Comment by Pictor13** [comment]
I think that if you want to transfer the Kanban (aka: _transfer the issues_ that got de-referenced with moving orgs):

- create a new template ("Make a copy", iirc) from the old Project & use it into the new org's Project.
- got to issues list & select all issues
- from the "Projects" dropdown select the new Organization & choose the new Project
- you now have the same issue under the new project.   

Unfortunately I couldn't find a way to maintain the distribution (and position) of the cards among the boards.

**Reply by ankitsejwal** [reply]
Thanks. You saved me a lot of time. I hope they add a transfer feature soon to those looking for it.

**Reply by RemDelaporteMathurin** [reply]
Unfortunately this doesn't transfer Issues and Pull requests. Thanks however!

**Comment by Mostafa33T** [comment]
Hello guys, 

Did you find a solution here ? I'm facing the same issue : I can transfer a repo from organisation A to organisation B, but the project (which is based on the issues of the repo) is not transferred from A to B.

Do you know how can I transfer the project (with issues and their distribution) in the new org ?

Thanks 🙏

**Reply by alertNiKole** [reply]
Hi 
I had the same issue tonight. Heres what I found:

**Scenario**
I created a GH Project under my personal account but needed to move it to one of my organizations.

Go to the Project Screen: Click the 3 dots on right and choose **"Make a Copy"**
**_Note: Issues and Pull requests cannot be copied_**, you can however copy draft issues over

![image](https://github.com/community/community/assets/28637298/642c4ea9-4a0c-43aa-93c7-fec0f83f3a97)

Choose the **New Owner**, for me this was my company

![image](https://github.com/community/community/assets/28637298/f6e960c1-d2d7-4d7c-93d1-a0c3c4194a41)


Rename the project as needed
![image](https://github.com/community/community/assets/28637298/37e40fcc-976d-4aed-9b77-b62fd7b9780b)

Copy when complete, it comes up pretty instantaneously and can be found under the new org
![image](https://github.com/community/community/assets/28637298/eb24bef4-172f-46c8-8c40-1714b3068257)

**Note:** After the copy is complete I did find that assignees need to be reallocated to existing tasks. There isn't a likely a good way to make this happen automatically cross - organization due to permissions and membership. So its something to be aware of. the Project used here wasn't robust and did not include workflow automation so there may be other things that don't transfer but it is defiantly better than a manual recreation. 

Hope this helps

**Comment by pombredanne** [comment]
Anyone would be aware of a command line script using the API that can make this copy correctly, including issues?

Not being able to copy issues items make the current copy feature mostly harmless in practice.

**Comment by pombredanne** [comment]
See also: https://github.com/orgs/community/discussions/52535#discussioncomment-10277540

**Comment by MichaMican** [comment]
Is there a possibility to make this* a feature request somewhere? 

*Adding support for migrating GitHub Projects across organizations including issues

**Comment by Xelliese** [comment]
GitHub doesn’t let you transfer projects between organizations directly, but you can clone the board and recreate it in the new org.

**Reply by pombredanne** [reply]
yeah, like by hand. Not really a solution

**Reply by carlspring** [reply]
Who was the nut-job who came up with this "solution"...? If I wanted to just create a new empty project, I would sure as hell not click the "Copy" button, but go and create a new empty one. BY HAND. Some "solution".

**Reply by MichaMican** [reply]
The Copy button does copy a few settings & config stuff (e.g. the Custom fields and the notes that didn't got converted to issues yet. So it does in fact has an usecase.
But i am with you that i would rather have a full clone feature for projects that also copy my issues over - especially because Projects with multi Org issues is a thing and does work already.

**Comment by pombredanne** [comment]
@MichaMican This is a problem solved by @keshav-space with this magnificent script https://gist.github.com/keshav-space/f0afa93cb6ec1b9a39299deef9d6858b

It transfers the project including all non-issue items and issues items :heart:

**Reply by MichaMican** [reply]
Thanks a lot! I was planning on cooking up something similar - glad i don't have to write that myself anymore. Thx @keshav-space

**Comment by kmaehashi** [comment]
Hi all, I've posted an article on transferring projects between organizations using [gh-migrate-project](https://github.com/timrogers/gh-migrate-project) by @timrogers. Hope this helps!
https://medium.com/@kmaehashi/transferring-github-projects-between-orgs-91dd1e259d5d

**Comment by chmolto** [comment]
For anyone that might need it, forked originals @keshav-space gist to also support migrating custom fields on the project and apply them to the issues as they were 😄 

https://gist.github.com/chmolto/757701b4221377332036c345cd6d44f0

**Reply by pombredanne** [reply]
@keshav-space gist eventually became https://github.com/aboutcode-org/github-import-issues-csv that I use to create issues (and migrate or copy issues and more between projects)

**Comment by h-vetinari** [comment]
This will become more important with the new repository limit of 100'000 repos per organisation. For example, [conda-forge](https://conda-forge.org) runs a large distribution where each package gets built in a separate repo (a "feedstock"), and we're already at 25k repos. The [announcement](https://github.blog/changelog/2025-03-27-repository-ownership-limits/) for the new limits says:
>  If you require more than 100,000 repositories, you can distribute ownership across multiple organizations, maintaining seamless operations.

Seamless like moving repos across orgs...
